extensions/security/runtime/src/main/java/io/quarkus/security/runtime/SecurityIdentityProxy.java

```diff
@@ -60,7 +61,13 @@ public Map<String, Object> getAttributes() {
 
     @Override
     public Uni<Boolean> checkPermission(Permission permission) {
-        return association.getIdentity().checkPermission(permission);
+        return association.getDeferredIdentity()
```

Comment:
Why this change has to be done ?

Reply:
`getIdentity` maye be blocking (please see `io.quarkus.security.runtime.SecurityIdentityAssociation#getIdentity` for one blocking implementation) and if user is using `checkPermission` instead of `checkPermissionBlocking`, he might be using it on IO thread, therefore he wants to avoid blocking. That's important when proactive auth is disabled. I forgot about this before, but if you think about it, this change is positive.

Reply:
Also, it can't really break anything, can it? In my eyes, this is how `checkPermission` method should look like at the first place.

Reply:
@michalvavrik Sure, I just needed to clarify, thanks